drivers/dma/dma_pl330.h

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2020 Broadcom
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef DMA_PL330_H
+#define DMA_PL330_H
+
+#include <drivers/dma.h>
+
+#define DT_DRV_COMPAT arm_dma_pl330
+/*
+ * Max burst length and max burst size for 32bit system with
+ * 128bit bus width for memory to memory data transfer
+ *
+ * Burst length is encoded in following format for pl330
+ * b0000 = 1 data transfer
+ * b0001 = 2 data transfers
+ * b0010 = 3 data transfers
+ * .
+ * .
+ * b1111 = 16 data transfers
+ *
+ * Burst size is encoded in following format for pl330
+ * b000 = 1 byte
+ * b001 = 2 bytes
+ * b010 = 4 bytes
+ * b011 = 8 bytes
+ * b100 = 16 bytes
+ * b101 = 32 bytes
+ * b110 = 64 bytes
+ * b111 = 128 bytes.
+ */
+#define MAX_BURST_LEN		0xf /* 16byte data */
+#define MAX_BURST_SIZE_LOG2	4
+
+/*
+ * PL330 works only on 4GB boundary.
+ * PL330 has 32bit registers for source and destination addresses
+ */
+#define PL330_MAX_OFFSET	0x100000000
+
+/* PL330 supports max 16MB dma based on AXI bus size */
+#define PL330_MAX_DMA_SIZE	0x1000000
+
+/* Maximum possible values for PL330 ucode loop counters */
+#define PL330_LOOP_COUNTER0_MAX	0x100
+#define PL330_LOOP_COUNTER1_MAX	0x100
+
+#define MAX_DMA_CHANNELS	DT_INST_PROP(0, dma_channels)
+
+#define DMAC_PL330_CS0		0x100
+#define DMAC_PL330_DBGSTATUS	0xd00
+#define DMAC_PL330_DBGCMD	0xd04
+#define DMAC_PL330_DBGINST0	0xd08
+#define DMAC_PL330_DBGINST1	0xd0c
+
+#define DMA_TIMEOUT_US		10
+#define CH_STATUS_MASK		0xf
+#define DATA_MASK		0xf
+
+#define DMA_INTSR1_SHIFT	24
+#define DMA_INTSR0_SHIFT	16
+#define DMA_INTSR0		0xa0
+#define DMA_SECURE_SHIFT	17
+#define DMA_CH_SHIFT		8
+
+#define LS_ICFG_DMAC_AXI_ADD_CONTROL_BASE	0x482f005c
```

Comment:
why is a specific address present in generic driver? Seems wrong.

Reply:
Its unused, will remove. Ack